griptape/tools/shopify_client/tool.py

```diff
@@ -0,0 +1,66 @@
+from textwrap import dedent
+from typing import Optional
+from urllib.parse import urljoin
+import schema
+from schema import Schema, Literal
+from attr import define, field
+from griptape.tools import BaseTool
+from griptape.utils.decorators import activity
+from griptape.artifacts import BaseArtifact, TextArtifact, ErrorArtifact
+
+
+@define
+class ShopifyClient(BaseTool):
+    storename: str = field(kw_only=True)
+    storefront_access_token: str = field(kw_only=True)
+    shopify_endpoint: str = field(default="myshopify.com/api/2023-07/graphql.json")
+
+    @activity(
+        config={
+            "description": "Can be used to search for shopify products",
+            "schema": Schema(
+                {
+                    Literal("first", description="How many products to retreive."): str,
+                }
+            ),
+        }
+    )
+    def search(self, params: dict) -> BaseArtifact:
+        from requests import post, exceptions
+
+        first = params.get("first", 10)
+
+        url = f"https://{self.storename}.{self.shopify_endpoint}"
```

Comment:
Can we use something like `RestApiClient._build_url`, so it doesn't break because of extra/missing slashes? May be abstract it into a `util`?

Reply:
Does `urljoin` help here? We don't really have a base url until we join `storename` + `schema_endpoint`.